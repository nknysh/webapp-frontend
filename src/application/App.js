/**
 * Copyright Â© 2018 Gigster, Inc. All Rights Reserved.
 *
 * This software and/or related documentation, is protected by copyright and
 * other intellectual property laws, and Gigster and/or its licensors, as
 * applicable, own all right, title and interest in and to its content, and all
 * derivatives, translations, adaptations and combinations of the foregoing. You
 * are not permitted to copy, distribute, transmit, store, display, perform,
 * reproduce, publish, license, create derivative works from, transfer, sell, or
 * make any other use of this software and/or related documentation unless you
 * have entered into a written agreement with Gigster regarding such usage. You
 * agree that all such usage of the software and/or related documentation shall
 * be subject to the terms and conditions of such written agreement, including
 * all applicable license restrictions.
 */

// Libraries
import React from 'react';
import { connect } from 'react-redux';

// App
import { getUserFromToken } from 'actions/auth';
import { AppRouter, getQuery } from '../router';

// Styles
import './App.css';

class App extends React.Component {
  getTokenParam = () => {
    if (!window) {
      return null;
    }

    const query = getQuery(window.location) || {};
    return query.token;
  }

  componentWillMount() {
    // TODO(mark): Probably not the best place but we don't have sagas.
    const tokenParam = this.getTokenParam();
    const token = tokenParam || localStorage.getItem('authToken');

    if (token) {
      this.props.getUserFromToken({ token });
    }
  }

  render() {
    return <AppRouter />;
  }
}

export default connect(undefined, { getUserFromToken })(App);
