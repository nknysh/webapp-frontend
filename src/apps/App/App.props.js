import PropTypes from 'prop-types';

import { noop } from 'utils';

export const propTypes = {
  setToken: PropTypes.func,
  isAppVersionDeprecated: PropTypes.bool,
};

export const defaultProps = {
  setToken: noop,
  isAppVersionDeprecated: false,
};
