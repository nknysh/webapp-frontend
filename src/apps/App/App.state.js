import { connect } from 'react-redux';
import { pipe } from 'ramda';
import { setToken } from 'store/modules/auth/actions';
import { resetStatuses } from 'store/common/actions';
import { getCurrentUser } from 'store/modules/auth/selectors';

export const mapStateToProps = state => ({
  currentUser: getCurrentUser(state),
});

export const mapDispatchToProps = dispatch => ({
  setToken: pipe(
    setToken,
    dispatch
  ),
  resetStatuses: pipe(
    resetStatuses,
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
);
