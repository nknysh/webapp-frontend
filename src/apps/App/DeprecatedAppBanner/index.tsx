import React from 'react';
import styled  from 'styled-components';

const DeprecatedAppBannerComponent = () =>  (
  <div>
    This version of app is deprecated. In order to avoid getting errors, please, reload page to get the latest version.
  </div>
);

const DeprecatedAppBanner = styled(DeprecatedAppBannerComponent)`
  width: 100%;
  background-color: yellow;
`;

export default DeprecatedAppBanner;
