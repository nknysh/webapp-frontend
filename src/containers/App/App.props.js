import PropTypes from 'prop-types';

export const propTypes = {
  getUserFromToken: PropTypes.func,
};
