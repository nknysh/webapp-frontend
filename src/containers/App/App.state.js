import { connect } from 'react-redux';
import { pipe } from 'ramda';
import { getUserFromToken } from 'actions/auth';

export const mapStateToProps = () => ({});

export const mapDispatchToProps = dispatch => ({
  getUserFromToken: pipe(
    getUserFromToken,
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
);
