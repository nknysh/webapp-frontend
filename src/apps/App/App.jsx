import React, { Suspense, useEffect } from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import { compose, prop } from 'ramda';
import { ErrorBoundary, Loader } from '@pure-escapes/webapp-ui-components';

import { getAppRoutes } from 'routing';

import { Layout } from 'components';
import { useScrollToTop, useEffectBoundary } from 'effects';
import { withUser } from 'hoc';

import { propTypes, defaultProps } from './App.props';
import connect from './App.state';
import { FastSearchContainerConnected } from 'containers/FastSearch';
import { RatesImportContainerConnected } from 'containers/RatesImport';
import { AllotmentsImportContainerConnected } from 'containers/AllotmentsImport';
import { OfferRouting } from 'containers/OfferRouting';
import { AuthRoute } from 'containers/AuthRoute';
import { EUserType } from 'services/BackendApi';

export const App = ({
  location: { pathname },
  user,
  resetStatuses,
  pageChange,
  bootstrapAppRequestAction,
  isAppVersionDeprecated,
}) => {
  // Scroll to top on path change
  useScrollToTop(pathname);

  useEffect(() => {
    bootstrapAppRequestAction();
  }, [bootstrapAppRequestAction]);

  useEffectBoundary(() => {
    // Trigger page change action on pathname change
    pageChange(pathname);

    // Reset statuses on page change
    resetStatuses();
  }, [pathname]);

  return (
    <Layout isAppVersionDeprecated={isAppVersionDeprecated}>
      <Suspense fallback={<Loader />}>
        <Switch>
          <Route path="/search/beta" exact component={FastSearchContainerConnected} />
          <Route path="/offers" component={OfferRouting} />
          <AuthRoute allow={[EUserType.RL, EUserType.ADMIN]} exact path="/rates/import" component={RatesImportContainerConnected} />
          <AuthRoute allow={[EUserType.RL, EUserType.ADMIN]} exact path="/allotments/import" component={AllotmentsImportContainerConnected} />
          {...getAppRoutes(prop('type', user))}
        </Switch>
      </Suspense>
    </Layout>
  );
};

App.propTypes = propTypes;
App.defaultProps = defaultProps;

export default compose(ErrorBoundary, withRouter, withUser, connect)(App);
