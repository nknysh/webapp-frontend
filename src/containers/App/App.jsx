import React, { PureComponent } from 'react';
import { Switch, BrowserRouter } from 'react-router-dom';
import { compose } from 'ramda';

import routes from 'routing';
import { ErrorBoundary } from 'hoc/ErrorBoundary';
import { getToken } from 'utils/auth';
import { windowExists } from 'utils/window';

import { propTypes } from './App.props';
import connect from './App.state';
import './App.css';

export class App extends PureComponent {
  componentDidMount() {
    const { getUserFromToken } = this.props;
    const token = getToken(windowExists);

    if (token) {
      getUserFromToken(token);
    }
  }

  render() {
    return (
      <BrowserRouter>
        <Switch>{routes}</Switch>
      </BrowserRouter>
    );
  }
}

App.propTypes = propTypes;

export default compose(
  ErrorBoundary,
  connect
)(App);
