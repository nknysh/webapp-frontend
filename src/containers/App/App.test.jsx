import React from 'react';

import { App } from './App';

const getShallow = props => shallow(<App getUserToken={() => {}} {...props} />);

describe('<App />', () => {
  describe('render', () => {
    it('matches snapshot', () => {
      const component = getShallow();
      expect(component).toMatchSnapshot();
    });
  });
});
