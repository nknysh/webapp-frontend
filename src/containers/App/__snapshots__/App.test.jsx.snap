// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render matches snapshot 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="2c68c93ab90d6c00df2b877d8b10fc27e9e69040"
      name="Root Path"
      path="/"
    />
    <Route
      component={[Function]}
      key="e0df44d8d54f1c044a98d9acc9023081729366b7"
      name="About Path"
      path="/about"
    />
    <Route
      component={[Function]}
      key="d01ff74abe0ad9b7fa7ab4e8b7461d623c9cff4c"
      name="Contact Path"
      path="/contact"
    />
    <Route
      component={[Function]}
      key="0fa7346eacc92bb8d7d490ebd39ae1ec19b34893"
      name="Privacy Path"
      path="/privacy"
    />
    <Route
      component={[Function]}
      key="f6ce7f3fc19f0177e19403a4c9d3c1ef48a3c864"
      name="Terms Path"
      path="/terms"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="74010ebd2b89f2f5882c2978a75b50094861c034"
      name="Calendar Path"
      path="/calendar"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="e2560e9b43f99eb42d6e798125f52a60823f39b3"
      name="Hotels Search Path"
      path="/hotels"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="77518f0ffed68a0def79b9644d89159e4580a669"
      name="Hotel View Path"
      path="/hotels/:id"
    />
    <Redirect
      exact={true}
      from="/admin"
      key="cb9d1398df770b9dcdfbff278b8ea27ee2f49068"
      name="Admin Path"
      push={false}
      to="/admin/dashboard"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="8b27d228d8817c59e95bff7d9469f951fe664216"
      name="Admin Dashboard Path"
      path="/admin/dashboard"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="e053ecfddaeb47739e6ef7ed0e25a777816b9a48"
      name="Admin Bookings List Path"
      path="/admin/bookings"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="0e2eb7416b96a29aa0d510ae2d582037a9505447"
      name="Admin Hotels List Path"
      path="/admin/hotels"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="06d836aed74c8433baf4953eafdf1206da148e22"
      name="Admin Hotel Create Path"
      path="/admin/hotels/new"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="6a0604ce1b371f3674e88ea87fbdda6621a4d6f6"
      name="Admin Hotel Update Path"
      path="/admin/hotels/:id/edit"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="452c8bbaff582820db4de780c69d72a78bbb9d07"
      name="Admin Room Create Path"
      path="/admin/hotels/:id/rooms/new"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="e7e0997783c3466096dbdd92d2e332b626d98558"
      name="Admin Room Update Path"
      path="/admin/hotels/:hotelId/rooms/:id/edit"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="aa70933bd0588337773bbbc3a22a93c5b009d76c"
      name="Admin Rates List Path"
      path="/admin/rooms/:id/rates"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="fd1a6edb7280b31d7cfc1013f2a63c51899f5981"
      name="Admin Proposals List Path"
      path="/admin/proposals"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="ce7cd33dce08a37eb88a6b518942377232c29dfa"
      name="Admin Sales Representatives List Path"
      path="/admin/sales-representatives"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="145b273ebb0ea8dcb0f55767a7f4fadba4d191b6"
      name="Admin Travel Agents List Path"
      path="/admin/travel-agents"
    />
    <Redirect
      exact={true}
      from="/admin"
      key="cb9d1398df770b9dcdfbff278b8ea27ee2f49068"
      name="Admin Path"
      push={false}
      to="/admin/dashboard"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="8b27d228d8817c59e95bff7d9469f951fe664216"
      name="Admin Dashboard Path"
      path="/admin/dashboard"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="e053ecfddaeb47739e6ef7ed0e25a777816b9a48"
      name="Admin Bookings List Path"
      path="/admin/bookings"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="0e2eb7416b96a29aa0d510ae2d582037a9505447"
      name="Admin Hotels List Path"
      path="/admin/hotels"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="06d836aed74c8433baf4953eafdf1206da148e22"
      name="Admin Hotel Create Path"
      path="/admin/hotels/new"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="6a0604ce1b371f3674e88ea87fbdda6621a4d6f6"
      name="Admin Hotel Update Path"
      path="/admin/hotels/:id/edit"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="452c8bbaff582820db4de780c69d72a78bbb9d07"
      name="Admin Room Create Path"
      path="/admin/hotels/:id/rooms/new"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="e7e0997783c3466096dbdd92d2e332b626d98558"
      name="Admin Room Update Path"
      path="/admin/hotels/:hotelId/rooms/:id/edit"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="aa70933bd0588337773bbbc3a22a93c5b009d76c"
      name="Admin Rates List Path"
      path="/admin/rooms/:id/rates"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="fd1a6edb7280b31d7cfc1013f2a63c51899f5981"
      name="Admin Proposals List Path"
      path="/admin/proposals"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="ce7cd33dce08a37eb88a6b518942377232c29dfa"
      name="Admin Sales Representatives List Path"
      path="/admin/sales-representatives"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="145b273ebb0ea8dcb0f55767a7f4fadba4d191b6"
      name="Admin Travel Agents List Path"
      path="/admin/travel-agents"
    />
    <Route
      component={[Function]}
      key="d814b5d29d2dcc4ec699c507b02b964640a9e3f7"
      name="Login Path"
      path="/login"
    />
    <Route
      component={[Function]}
      key="df87dfd9f070ecaa9cfd68a0f6ec01ee33f3f266"
      name="Reset Password Path"
      path="/password/reset"
    />
    <Route
      component={[Function]}
      key="c966e8f3324076a3971126723616b5a6356e2c7d"
      name="Set Password Path"
      path="/password/new"
    />
    <Route
      component={[Function]}
      key="f728f3fd178d2d4d7178691710943d2105855297"
      name="Create User Path"
      path="/sign-up"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="eb545fd0536b6f81bdba0d6b4c5e786a02139d8b"
      name="Bookings Path"
      path="/bookings"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="5092870766a8e801f34a9456de0f4bda731fea6e"
      name="Booking Guest Details Path"
      path="/bookings/:id/guests"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="eb01b0467c49325b50e4eb73d94e15178b527f06"
      name="Booking Payment Path"
      path="/bookings/:id/payment"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="5a32ce59fa99412d19ab1b5e776d3f25431641aa"
      name="Booking View Path"
      path="/bookings/:id"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="cf8c03454529c2d83f342fcf11b49c0a836adf92"
      name="Booking Options (Holds) Path"
      path="/holds"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="c62f889f908bd02993eb938d9927ce3c0293fe6d"
      name="Proposal View Path"
      path="/proposals"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="72f2ec16c463fff6e66c3c09dff84dfc51d8ae41"
      name="Proposal View Path"
      path="/proposals/:id"
    />
    <Redirect
      exact={true}
      from="/settings"
      key="3ddad6890ec5779cb00843d3c1a9c66827cb55ec"
      name="Settings Path"
      push={false}
      to="/settings/profile"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="d1953d07864292bec02f2188f55f002b1fdc3d33"
      name="Settings Company Path"
      path="/settings/company"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="964e66a50dd2087782368e1688d97e4dfee818d2"
      name="Settings Notifications Path"
      path="/settings/notifications"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="d531c6f0290fcb468a753aa49400c390f3b4d00b"
      name="Settings Preferences Path"
      path="/settings/preferences"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="6645443eb8a6091435c09a5127af32ace98aaac5"
      name="Settings Profile Path"
      path="/settings/profile"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="be8b4d3ba89cdfe8fae1a2db9415d113c1a374cd"
      name="Settings Travel Agents Path"
      path="/settings/travel-agents"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="ace675a3349a0e18781765e98353cb17e6f4f637"
      name="Travel Agent View Path"
      path="/users/:id"
    />
    <withAuthentication(withRouter(AuthenticatedRoute))
      component={[Function]}
      key="1ecb1828c8aa13c20bab9be95a49b8fff01f93d7"
      name="User Bookings Path"
      path="/users/:id/bookings"
    />
    <Route
      component={[Function]}
      key="ae4ef0dc5812ee3f8bc7fbd3728741c4b0620c4b"
      name="Not Found"
    />
  </Switch>
</BrowserRouter>
`;
